### ISSUE TITLE
SickChill failed to update on Ubuntu 18.04, now won't start.

### ISSUE THREAD

**Issue by hapklaar** [SickChill/SickChill#7412]
### Before submitting your issue:

Enable debug logging in SickChill settings, reproduce the error (be sure to disable after the bug is fixed)

Branch/Commit:
cur_commit = 27701a96bf26a5ba2d5a5bc45ab41b661b8bf4bb
newest_commit = 5f63ed027cd80bd85604885d67d3fef3e40f42c9
num_commits_behind = 206
OS: Ubuntu 18.04.5
Browser: MS Edge
What you did: Update from GUI to latest commit
What happened: Update failed with a mako error, SC won't start anymore
What you expected: Successful update
Logs:
```
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: Checking if we can use git commands: git version
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: Executing git version with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: git version : returned successful
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: Using: git
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: Executing git symbolic-ref -q HEAD with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: git symbolic-ref -q HEAD : returned successful
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: Executing git symbolic-ref -q HEAD with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: git symbolic-ref -q HEAD : returned successful
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: Executing git rev-parse HEAD with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: git rev-parse HEAD : returned successful
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: Executing git config remote.origin.url https://github.com/SickChill/SickChill.git with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: git config remote.origin.url https://github.com/SickChill/SickChill.git : returned successful
2021-08-08 20:26:31 DEBUG :: WEBSERVER-HOME_0 :: Executing git fetch origin --prune with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: git fetch origin --prune : returned successful
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: Executing git branch --set-upstream-to origin/master with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: git branch --set-upstream-to origin/master : returned successful
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: Executing git rev-parse --verify --quiet "@{upstream}" with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: git rev-parse --verify --quiet "@{upstream}" : returned successful
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: Executing git rev-list --left-right "@{upstream}"...HEAD with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: git rev-list --left-right "@{upstream}"...HEAD : returned successful
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: cur_commit = 27701a96bf26a5ba2d5a5bc45ab41b661b8bf4bb, newest_commit = 5f63ed027cd80bd85604885d67d3fef3e40f42c9, num_commits_behind = 206, num_commits_ahead = 0
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: Executing git config remote.origin.url https://github.com/SickChill/SickChill.git with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: git config remote.origin.url https://github.com/SickChill/SickChill.git : returned successful
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: Executing git reset --hard with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: git reset --hard : returned successful
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: Executing git symbolic-ref -q HEAD with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: git symbolic-ref -q HEAD : returned successful
2021-08-08 20:26:32 DEBUG :: WEBSERVER-HOME_0 :: Executing git pull -f origin master with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:34 DEBUG :: WEBSERVER-HOME_0 :: git pull -f origin master : returned successful
2021-08-08 20:26:34 DEBUG :: WEBSERVER-HOME_0 :: Executing git rev-parse HEAD with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:34 DEBUG :: WEBSERVER-HOME_0 :: git rev-parse HEAD : returned successful
2021-08-08 20:26:34 DEBUG :: WEBSERVER-HOME_0 :: Executing git clean -df lib with your shell in /opt/sickchill/sickchill
2021-08-08 20:26:34 DEBUG :: WEBSERVER-HOME_0 :: git clean -df lib : returned successful
2021-08-08 20:26:34 DEBUG :: WEBSERVER-HOME_0 :: Sending notification for SickChill Updated To Commit#: 5f63ed027cd80bd85604885d67d3fef3e40f42c9
2021-08-08 20:26:34 DEBUG :: WEBSERVER-HOME_0 :: Pushover API KEY in use: ########
2021-08-08 20:26:34 INFO :: WEBSERVER-HOME_0 :: A mako error occurred: Could not find a suitable TLS CA certificate bundle, invalid path: /opt/sickchill/lib3/certifi/cacert.pem
```

This file appears to have been deleted by the update process, as it was there before starting the update. (restored vm snapshot to check)

Now when I try to start sickchill it fails:

```
root@haplxc-PMS:/opt/sickchill/Logs# /usr/bin/python3 /opt/sickchill/SickChill.py -q --nolaunch --datadir=/opt/sickchill
Checking poetry
Can write to /usr/local/lib/python3.6/dist-packages: True
Can write to /root/.local/lib/python3.6/site-packages: False
We always run from virtualenv when running from source
Restarting SickChill with ['/opt/sickchill/.venv/bin/python3.6', '/opt/sickchill/SickChill.py', '-q', '--nolaunch', '--datadir=/opt/sickchill']
Checking poetry
sickchill found: False
Can write to /root/.local/lib/python3.6/site-packages: False
Can write to /opt/sickchill/.venv/lib/python3.6/site-packages: True
pip found: True
pip args: /opt/sickchill/.venv/bin/python3.6 -m pip install --no-input --disable-pip-version-check --no-python-version-warning --no-color -qU --find-links=https://wheel-index.linuxserver.io/ubuntu/ setuptools poetry poetry-date-version-plugin wheel --pre
poetry found: True
pip args: /opt/sickchill/.venv/bin/python3.6 -m pip install --no-input --disable-pip-version-check --no-python-version-warning --no-color -qU --find-links=https://wheel-index.linuxserver.io/ubuntu/ Warning: The lock file is not up to date with the latest changes in pyproject.toml. You may be getting outdated dependencies. Run update to update them. appdirs==1.4.4 babelfish==0.6.0b1; python_version >= "3.6" and python_version < "4.0" beautifulsoup4==4.9.3 beekeeper-alt==2021.7.16; python_full_version >= "3.6.2" and python_version < "4" bencode.py==4.0.0 cachecontrol==0.12.6; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.4.0") certifi==2021.5.30; python_version >= "2.7" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" cffi==1.14.6; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6" or python_full_version >= "3.5.0" and python_version >= "3.6" chardet==4.0.0; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.5.0") charset-normalizer==2.0.3; python_full_version >= "3.6.0" and python_version >= "3" click==8.0.1; python_version >= "3.6" cloudscraper==1.2.58 colorama==0.4.4; python_version >= "3.6" and python_full_version < "3.0.0" and platform_system == "Windows" or platform_system == "Windows" and python_version >= "3.6" and python_full_version >= "3.5.0" configobj==5.0.6 cryptography==3.4.7; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" and python_version >= "3.6" decorator==5.0.9; python_version >= "3.5" deluge-client==1.9.0 deprecated==1.2.12; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6" dogpile.cache==1.1.3; python_version >= "3.6" enzyme==0.4.1 feedparser==6.0.8; python_version >= "3.6" future==0.18.2; python_version >= "2.6" and python_full_version < "3.0.0" or python_full_version >= "3.3.0" gntp==1.0.3 greenlet==1.1.0; python_version >= "3" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version >= "3" guessit==3.3.1 html5lib==1.1; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.5.0") httplib2==0.19.1 idna==3.2; python_version >= "3.5" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version >= "3.5" ifaddr==0.1.7 imagesize==1.2.0; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.4.0") imdbpy==2021.4.18 importlib-metadata==1.7.0; python_version >= "3.6" and python_full_version < "3.0.0" and python_version < "3.8" or python_version < "3.8" and python_version >= "3.6" and python_full_version >= "3.5.0" or python_version >= "2.7" and python_full_version < "3.0.0" and python_version < "3.8" or python_full_version >= "3.6.0" and python_version < "3.8" ipaddress==1.0.23 js2py==0.71 jsonrpclib-pelix==0.4.2 kodipydent-alt==2021.7.16; python_full_version >= "3.6.2" and python_version < "4" lxml==4.6.3; python_version >= "2.7" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" mako==1.1.4; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.4.0") markdown2==2.4.0; python_version >= "3.5" and python_version < "4" markupsafe==2.0.1; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6" msgpack==1.0.2; python_version >= "2.7" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" new-rtorrent-python==1.0.1a0 oauthlib==3.1.1; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6" packaging==20.9; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.4.0") pbr==5.6.0; python_version >= "3.6" profilehooks==1.12.0 putio.py==8.7.0 pyaes==1.6.1 pycparser==2.20; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6" pygithub==1.55; python_version >= "3.6" pyjsparser==2.7.1 pyjwt==2.1.0; python_version >= "3.6" pymediainfo==5.1.0; python_version >= "3.6" pynacl==1.4.0; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6" pynma==1.0 pyopenssl==20.0.1; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.5.0") pyparsing==2.4.7; python_version >= "2.6" and python_full_version < "3.0.0" or python_full_version >= "3.3.0" or python_version >= "2.7" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" pysocks==1.7.1; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.4.0") pysrt==1.1.2 python-dateutil==2.8.2; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.3.0") python-slugify==5.0.2; python_version >= "3.6" python-twitter==3.5 python3-fanart==2.0.0 pytz==2021.1 qbittorrent-api==2021.5.22 rarfile==4.0 rebulk==3.0.1 requests-oauthlib==1.3.0; python_version >= "2.7" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" requests-toolbelt==0.9.1 requests==2.26.0; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.6.0") send2trash==1.7.1 sgmllib3k==1.0.0; python_version >= "3.6" six==1.16.0; python_version >= "2.7" and python_full_version < "3.0.0" or python_full_version >= "3.3.0" or python_full_version >= "3.5.0" or python_version >= "3.4" and python_full_version < "3.0.0" or python_full_version >= "3.3.0" and python_version >= "3.4" or python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6" soupsieve==2.2.1; python_version >= "3.6" sqlalchemy==1.4.22; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.6.0") stevedore==3.3.0; python_version >= "3.6" subliminal==2.1.0 text-unidecode==1.3; python_version >= "3.6" timeago==1.0.15 tmdbsimple==2.8.0 tornado==6.1; python_version >= "3.5" tus.py==1.3.4 tvdbsimple==1.0.6 tzlocal==2.1 unidecode==1.2.0; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.4.0") urllib3==1.26.6; python_version >= "2.7" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" and python_version < "4" or python_full_version >= "3.6.0" and python_version < "4" validators==0.18.2; python_version >= "3.4" webencodings==0.5.1; python_version >= "2.7" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" win-inet-pton==1.1.0 wrapt==1.12.1; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6" xmltodict==0.12.0; python_full_version >= "3.6.2" and python_version < "4" zipp==3.5.0; python_version >= "3.6" and python_full_version < "3.0.0" and python_version < "3.8" or python_version < "3.8" and python_version >= "3.6" and python_full_version >= "3.5.0"
Command result: ERROR: Invalid requirement: 'Warning: The lock file is not up to date with the latest changes in pyproject.toml. You may be getting outdated dependencies. Run update to update them.'

Trying user site-packages
Command result: ERROR: Can not perform a '--user' install. User site-packages are not visible in this virtualenv.

Traceback (most recent call last):
  File "/opt/sickchill/SickChill.py", line 14, in <module>
    import sickchill.start
  File "/opt/sickchill/sickchill/__init__.py", line 6, in <module>
    from .show.indexers import indexer, ShowIndexer
  File "/opt/sickchill/sickchill/show/indexers/__init__.py", line 1, in <module>
    from .handler import ShowIndexer
  File "/opt/sickchill/sickchill/show/indexers/handler.py", line 3, in <module>
    from sickchill import logger, settings
  File "/opt/sickchill/sickchill/logger.py", line 12, in <module>
  File "/opt/sickchill/sickchill/logger.py", line 12, in <module>
    from github import InputFileContent
ModuleNotFoundError: No module named 'github'
```

**Comment by jallmantalbot** [comment]
I had exactly the same error, and fixed it by running:

`/opt/SickChill/.venv/bin/poetry update`

Not sure exactly what's causing this, but after running that (and it ran for a good 4-5 minutes) SickChill started successfully.

**Comment by potequity** [comment]
Running 'poetry' did not fix it for me, here's the output:

sickchill@linux02:~$ /opt/sickchill/.venv/bin/poetry update

  RuntimeError

  Poetry could not find a pyproject.toml file in /var/lib/sickchill or its parents

  at /opt/sickchill/.venv/lib/python3.9/site-packages/poetry/core/factory.py:435 in locate
      431│             if poetry_file.exists():
      432│                 return poetry_file
      433│ 
      434│         else:
    → 435│             raise RuntimeError(
      436│                 "Poetry could not find a pyproject.toml file in {} or its parents".format(                                                     
      437│                     cwd
      438│                 )
      439│             )
sickchill@linux02:~$

**Comment by hapklaar** [comment]
@jallmantalbot thanks, that fixed it for me, almost.

I'm now able to start SC by running 
```
/usr/bin/python3 /opt/sickchill/SickChill.py -q --nolaunch --datadir=/opt/sickchill
```

But when using systemctl, it fails without any error:

```
root@haplxc-PMS:/opt/sickchill/Logs# systemctl status sickchill.service 
● sickchill.service - SickChill Daemon
   Loaded: loaded (/etc/systemd/system/sickchill.service; enabled; vendor preset: enabled)
   Active: inactive (dead) since Mon 2021-08-09 12:17:00 CEST; 1min 25s ago
  Process: 19347 ExecStart=/usr/bin/python3 /opt/sickchill/SickChill.py -q --daemon --nolaunch --datadir=/opt/sickchill (code=exited, status=0/SUCCESS)

Aug 09 12:17:00 haplxc-PMS systemd[1]: Starting SickChill Daemon...
Aug 09 12:17:00 haplxc-PMS systemd[1]: Started SickChill Daemon.
````

**Comment by hapklaar** [comment]
Fixed it by correcting permissions on the .venv dir:

```
chown -R sickchill:sickchill .venv/
```

Now all is well again.

**Comment by hapklaar** [comment]
@miigotu seems the upgrade process is not working at it should, do you want to keep this open for investigation?

**Comment by potequity** [comment]
Changing the ownership of /opt/sickchill/.venv did not make a difference, it still fails to start. Here's the ls output of /opt/sickchill/.venv

root@linux02:/opt/sickchill/.venv# ls -al
total 8
drwxrwxr-x 5 sickchill sickchill   74 Aug  8 15:03 .
drwxr-xr-x 8 sickchill sickchill 4096 Aug  8 15:03 ..
drwxrwxrwx 2 sickchill sickchill  201 Aug  8 22:51 bin
drwxrwxrwx 2 sickchill sickchill    6 Aug  8 15:03 include
drwxrwxrwx 3 sickchill sickchill   23 Aug  8 15:03 lib
lrwxrwxrwx 1 sickchill sickchill    3 Aug  8 15:03 lib64 -> lib
-rw-rw-rw- 1 sickchill sickchill   69 Aug  8 15:03 pyvenv.cfg
root@linux02:/opt/sickchill/.venv#

-PE

**Comment by potequity** [comment]
[sickchill-start-errors.txt](https://github.com/SickChill/SickChill/files/6955316/sickchill-start-errors.txt)

**Comment by hapklaar** [comment]
@potequity You might be better off starting a new issue as your problem is different from ours.

**Comment by miigotu** [comment]
> [sickchill-start-errors.txt](https://github.com/SickChill/SickChill/files/6955316/sickchill-start-errors.txt)
> 
> 

You have to update using the poetry inside the venv.

```bash
cd /opt/sickchill
.venv/bin/poetry update
```

**Comment by miigotu** [comment]
> @miigotu seems the upgrade process is not working at it should, do you want to keep this open for investigation?

Honestly somehow my lock file just got out of sync. Someone (or me when I can get to a PC) could check out a clean source, and run:
(Change python3.7 to what version of python3.6+ you use)

```bash
cd path/to/sickchill
python -m venv -p python3.7 .venv
source .venv/bin/activate
pip install poetry --pre
# make a new branch for a pr
git checkout -b update-poetry-lock
poetry update
poetry install --no-root --no-dev
 
# make sure it's just poetry.lock changed
git status
git commit -asm "Update poetry lock file"
# make a pr to the develop branch
```

**Comment by potequity** [comment]
Thanks, the '.venv/bin/poetry update' ran and completed without any errors. 
As for starting the service when I try to 'systemctl start sickchill' it gives me an 'Access Denied' error message. But not surprisingly it does start successfully when I run 'systemctl start sickchill' as root.
I'm now going to see if the service starts upon reboot, will post the results after the reboot is done......

I'd like to help out with the lock issue but the problem is I don't have another system laying around. I suppose I could look into registering for a free VM somewhere but that could take time.

**Comment by enormouspoon** [comment]
Just came here to say running: 
`
.venv/bin/poetry update
`
from within /opt/sickchill fixed my instance not starting. To avoid this in the future I guess I should update more frequently so I don't get 206 commits behind!

Thanks for all the help here folks.

**Comment by wdedood** [comment]
> > [sickchill-start-errors.txt](https://github.com/SickChill/SickChill/files/6955316/sickchill-start-errors.txt)
> 
> You have to update using the poetry inside the venv.
> 
> ```shell
> cd /opt/sickchill
> .venv/bin/poetry update
> ```

Thanks, that was the solution!

**Comment by potequity** [comment]
Anyone tested this solution in Ubuntu Server 21.04? Because the service doesn't start even after running the 'poetry update' command.

Help please!

**Comment by potequity** [comment]
I manually started it after doing the update and it started without any issues. Did a reboot to test if it persists and it looks like it doesn't. 
And now it doesn't start manually either.

Yikes!

**Comment by potequity** [comment]
BTW, this is a new/clean install. Followed instructions in this document https://github.com/SickChill/SickChill/wiki/SickChill-installation-Ubuntu-15

# note: python already at version 3.9.5 so omitted that in the command below:

apt-get update && sudo apt-get install unrar-free git-core openssl libssl-dev

groupadd -g 3000 sickchill

useradd -c "Sickchill User Id" -s /bin/bash -d /var/lib/sickchill -u 3000 -g sickchill -m sickchill
usermod -aG sudo sickchill

passwd sickchill

su - sickchill

sudo mkdir /opt/sickchill 
sudo chown sickchill:sickchill /opt/sickchill
sudo -u sickchill git clone https://github.com/SickChill/SickChill.git /opt/sickchill

sudo cp -v /opt/sickchill/contrib/runscripts/init.systemd /etc/systemd/system/sickchill.service

sudo chown root:root /etc/systemd/system/sickchill.service
sudo chmod 644 /etc/systemd/system/sickchill.service

sudo apt install python3-pip

sudo systemctl start sickchill # to create the .venv dir in /opt/sickchill

cd /opt/sickchill
.venv/bin/poetry update

#Enable, start, and then check the status of your new service

sudo systemctl enable sickchill
sudo systemctl start sickchill
sudo systemctl status sickchill

**Comment by gdeans** [comment]
I had this problem on my debian 10 system so I built a new VM to test and updated python to 3.9.11 (current).  The curl script failed b/c it wanted to install python 3.8 so I installed using the guide and the manual steps.  There is no poetry file in /opt/sickchill/.venv/bin so I can't run the update.  Why does a new vanilla install fail like this and how do I fix it?

**Comment by miigotu** [comment]
Run SickChill.py, it installs poetry itself on first run.

**Comment by ggez89** [comment]
> I had this problem on my debian 10 system so I built a new VM to test and updated python to 3.9.11 (current). The curl script failed b/c it wanted to install python 3.8 so I installed using the guide and the manual steps. There is no poetry file in /opt/sickchill/.venv/bin so I can't run the update. Why does a new vanilla install fail like this and how do I fix it?

I tried that and get this:


```
pi@RAPTURE:/ $ /opt/SickChill/.venv/bin/poetry update
-bash: /opt/SickChill/.venv/bin/poetry: No such file or directory
```

I tried installing pip with curl
```
pi@RAPTURE:/opt/sickchill $ curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
curl: (7) Failed to connect to raw.githubusercontent.com port 443: Connection timed out
```

Then I tried installing with pip
```
pip install --user poetry

pi@RAPTURE:/opt/sickchill $ poetry --version
-bash: poetry: command not found
```

Once again sickrage is broken due to an update. Cant even reinstall from git

**Comment by kyuuk** [comment]
@ggez89 i guess you are trying to run sickchill with python 2, while sickchill supports only python 3.6+, try running it with python3 
`python3 /opt/sickchill/SickChill.py`

**Comment by kyuuk** [comment]
@ggez89  use the init scripts from the sickchill repo found here https://github.com/SickChill/SickChill/tree/master/contrib/runscripts

**Comment by kyuuk** [comment]
@gdeans  a quick google search leads to [this](https://askubuntu.com/questions/1239829/modulenotfounderror-no-module-named-distutils-util) and the solution is 
`sudo apt-get install python3-distutils`

that should fix everything and poetry will be installed

**Comment by ggez89** [comment]
> @ggez89 use the init scripts from the sickchill repo found here https://github.com/SickChill/SickChill/tree/master/contrib/runscripts

Nope, that doesn't work. I am using a clone of the git repository and used those scripts. I also tried using a fresh one from here just for fun. Still nothing.

**Comment by kyuuk** [comment]
@ggez89  i will need a little more details of what you did, cause if u followed the guide you mentioned it's pretty outdated, so 

what script are you using? 
what is the contenent of you file /etc/default/sickrage or /etc/default/sickchill ?

**Comment by ggez89** [comment]
I tried init.debian, and init.ubuntu. I am on raspbian latest, and previously init.ubuntu worked for me.

I followed the guide substituting sickrage for sickchill. Then I did the solution in this thread

```
cd /opt/sickchill
.venv/bin/poetry update
```

Sickchill launches when I run it from

`python3 sickchill.py`

but not when I try to run it from startup scripts.

**Comment by kyuuk** [comment]
@ggez89 if it launches, then the problem is with the startup script, check the file `/etc/default/sickchill `

**Comment by ggez89** [comment]
> @ggez89 if it launches, then the problem is with the startup script, check the file `/etc/default/sickchill `

That's not helpful, I'm literally using the files in this repo and it doesn't work. It used to work until someone pushed a broken commit. 

I updated SC via the updater and it broke. I purged it and reinstalled it from git and it doesn't work.  You see the problem?

**Comment by tkmcmaster** [comment]
I've been having the same problem for a while now too. The problem seems to be if you add the `--daemon` parameter. If I add that manually, It immediately exits. If I remove that parameter, it works. The default start-up scripts include that parameter.

**Comment by joshbaptiste** [comment]
How I fixed this error on my Debian 10 buster system...

1) /opt/sickchill/.venv/bin/poetry update
It failed with the error 
```
AttributeError                                                                                                                                              
  'EmptyConstraint' object has no attribute 'allows'  
```             
2) I then performed the hack patch given in https://github.com/python-poetry/poetry/issues/4410

3) reran /opt/sickchill/.venv/bin/poetry update

4) Sickchill now starts correctly

**Comment by miigotu** [comment]
The emptyconstraint order, and the lockfile error are fixed in the develop develop branch, you need poetry==1.1.8 for those fixes (idk why poetry has an alpha with a higher version number that is older than 1.1.8 but whatever)

Also, for Ubuntu and debian that use systemd, use init.systemd.

Finally, of you have a problem with how I run this project, you are mad about a bug and wanna talk shit, don't use it. This project is not for the ungrateful people who think they are entitled to my work. I don't have to do this at all, and I don't work for you. I do this for fun. When you make it NOT FUN, I don't do it. So keep your negative bullshit entitled little bitch comments to yourself. If this fits, you know it's you who I'm talking about. If you want something you can bitch about reliability or whatever pay for cable, Netflix, Amazon Prime, Disney+, HBO Max, Showtime, etc. If you have to pay for it you can complain, of not then you can't. You could always just download videos on your own and complain to yourself or write your own software. To everyone else who has dealt with this being a pain in the ass at times without crying and whining and being disrespectful and has tried to help out please ignore this part.

**Comment by ggez89** [comment]
> I've been having the same problem for a while now too. The problem seems to be if you add the `--daemon` parameter. If I add that manually, It immediately exits. If I remove that parameter, it works. The default start-up scripts include that parameter.

Thanks, you are the real MVP.

A competent dev would figured out when this issue started and see what commits could have broken it, instead of calling his users "entitled little bitch"s.

**Comment by miigotu** [comment]
> > I've been having the same problem for a while now too. The problem seems to be if you add the `--daemon` parameter. If I add that manually, It immediately exits. If I remove that parameter, it works. The default start-up scripts include that parameter.
> 
> Thanks, you are the real MVP.
> 
> A competent dev would figured out when this issue started and see what commits could have broken it, instead of calling his users "entitled little bitch"s.

Go pound sand. A competent dev who doesn't work for you and doesn't owe you shit doesn't answer to you or have to present themselves how you see fit.

**Comment by miigotu** [comment]
How some people don't understand that if I didn't do this they wouldn't have the software at all to begin with, and if I stop it won't work for very long and they still won't have it really trips me out. Like, let's talk shit and make the one person that's making shit for you for free not want to do it, and you can't make them do it... It still isn't fixed because of exactly the people like this. I'll stay on hiatus until the entitled brat dicks go away. It's one thing to want something for free. It's a whole different thing to demand it and demand someone work for you for free while also being a total asshole to them.

**Comment by kyuuk** [comment]
@miigotu  you should not even bother answering people like this, they can't understand anything, all they know is complaining.
just don't waste your time, let's keep it a fun project and a fun community, so just let this kind talk alone 😄

**Comment by ggez89** [comment]
It's funny how these sickbeard clones always implode after the egomaniac dev throws a temper tantrum 😁

I'm out, later nerds

**Comment by miigotu** [comment]
I really try not to block people, but there are now 3 total people blocked from the org.

**Comment by potequity** [comment]
@miigotu Don't mean to preach but if I were you, I wouldn't let any of these ungrateful idiots get to you! There are more of us who realize the value of what you're doing (for free!!!) and I personally am very grateful to all the support I get for SickChill. Forget the naysayers and SickChillax :-)

@hapklaar I'll open a new thread for my original issue.

Keep up the good work guys! And don't take shit from anyone!

-PE

**Comment by damagu** [comment]
I also don't seem to be able to get sickchill to start after attempting to update it and tried looking for a way to fix it which is how I ended up here. After a lot of mucking around with pips and python I managed to get poetry update to run but I got this error:
```
$ /opt/sickchill/.venv/bin/poetry update
Updating dependencies
Resolving dependencies... (1087.2s)

Writing lock file

  AttributeError

  'EmptyConstraint' object has no attribute 'allows'

  at .venv/lib/python3.7/site-packages/poetry/core/version/markers.py:290 in validate
      286│ 
      287│         if self._name not in environment:
      288│             return True
      289│ 
    → 290│         return self._constraint.allows(self._parser(environment[self._name]))
      291│ 
      292│     def without_extras(self) -> MarkerTypes:
      293│         return self.exclude("extra")
      294│ 
```
I saw @joshbaptiste's comment regarding the above error, looked at the link he provided and tried making the same changes to the markers.py file.
This is my diff output:
```
$ diff -uw /opt/sickchill/.venv/lib/python3.7/site-packages/poetry/core/version/markers.py{~,}
--- /opt/sickchill/.venv/lib/python3.7/site-packages/poetry/core/version/markers.py~       2021-09-11 12:57:30.401359387 +0800
+++ /opt/sickchill/.venv/lib/python3.7/site-packages/poetry/core/version/markers.py        2021-09-11 13:01:46.535715984 +0800
@@ -287,6 +287,9 @@
         if self._name not in environment:
             return True
 
+        if not hasattr(self._constraint, "allows"):
+            return True
+
         return self._constraint.allows(self._parser(environment[self._name]))
 
     def without_extras(self) -> MarkerTypes: 
```
By the looks of it my markers.py file was not identical to the one shown in the link to begin with because: 
1) the line number being changed in the linked version was 288 wheresa mine shows it as 287 (of course, that might just be an additional blank line somewhere but I don't know).
and 
2) the last line of code shown in the linked diff output is shown as:
```def without_extras(self):  # type: () -> MarkerTypes```
...whereas mine shows that line as:
```def without_extras(self) -> MarkerTypes:```  

Anyway... I tried running the poetry update again and got a whole mess of different errors:

```
$ /opt/sickchill/.venv/bin/poetry update
Updating dependencies
Resolving dependencies... (95.5s)

Package operations: 122 installs, 1 update, 0 removals

  • Installing smmap (4.0.0)
  • Installing gitdb (4.0.7)
  • Installing mccabe (0.6.1)
  • Installing multidict (5.1.0): Failed

  CalledProcessError

  Command '['/opt/sickchill/.venv/bin/python', '/opt/sickchill/.venv/lib/python3.7/site-packages/virtualenv/seed/wheels/embed/pip-21.0.1-py3-none-any.whl/pip', 'install', '--use-pep517', '--disable-pip-version-check', '--prefix', '/opt/sickchill/.venv', '--no-deps', '/root/.cache/pypoetry/artifacts/4d/37/78/a987ae790fd0ae8d61b254b37394e654ed19197059b0f17fd9ca2f4f82/multidict-5.1.0.tar.gz']' returned non-zero exit status 1.

  at /usr/lib/python3.7/subprocess.py:487 in run
       483│             raise
       484│         retcode = process.poll()
       485│         if check and retcode:
       486│             raise CalledProcessError(retcode, process.args,
    →  487│                                      output=stdout, stderr=stderr)
       488│     return CompletedProcess(process.args, retcode, stdout, stderr)
       489│ 
       490│ 
       491│ def list2cmdline(seq):

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/opt/sickchill/.venv/bin/python', '/opt/sickchill/.venv/lib/python3.7/site-packages/virtualenv/seed/wheels/embed/pip-21.0.1-py3-none-any.whl/pip', 'install', '--use-pep517', '--disable-pip-version-check', '--prefix', '/opt/sickchill/.venv', '--no-deps', '/root/.cache/pypoetry/artifacts/4d/37/78/a987ae790fd0ae8d61b254b37394e654ed19197059b0f17fd9ca2f4f82/multidict-5.1.0.tar.gz'] errored with the following return code 1, and output: 
  Processing /root/.cache/pypoetry/artifacts/4d/37/78/a987ae790fd0ae8d61b254b37394e654ed19197059b0f17fd9ca2f4f82/multidict-5.1.0.tar.gz
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'done'
      Preparing wheel metadata: started
      Preparing wheel metadata: finished with status 'done'
  Building wheels for collected packages: multidict
    Building wheel for multidict (PEP 517): started
    Building wheel for multidict (PEP 517): finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /opt/sickchill/.venv/bin/python /tmp/tmp761dqmcs build_wheel /tmp/tmpqdraunk2
         cwd: /tmp/pip-req-build-1d7hsga9
    Complete output (48 lines):
    *********************
    * Accelerated build *
    *********************
    running bdist_wheel
    running build
    running build_py
    creating build
    creating build/lib.linux-aarch64-3.7
    creating build/lib.linux-aarch64-3.7/multidict
    copying multidict/_multidict_base.py -> build/lib.linux-aarch64-3.7/multidict
    copying multidict/_abc.py -> build/lib.linux-aarch64-3.7/multidict
    copying multidict/_multidict_py.py -> build/lib.linux-aarch64-3.7/multidict
    copying multidict/__init__.py -> build/lib.linux-aarch64-3.7/multidict
    copying multidict/_compat.py -> build/lib.linux-aarch64-3.7/multidict
    running egg_info
    writing multidict.egg-info/PKG-INFO
    writing dependency_links to multidict.egg-info/dependency_links.txt
    writing top-level names to multidict.egg-info/top_level.txt
    reading manifest file 'multidict.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files found matching 'multidict/_multidict.html'
    warning: no previously-included files found matching 'multidict/*.so'
    warning: no previously-included files found matching 'multidict/*.pyd'
    warning: no previously-included files found matching 'multidict/*.pyd'
    no previously-included directories found matching 'docs/_build'
    adding license file 'LICENSE'
    writing manifest file 'multidict.egg-info/SOURCES.txt'
    copying multidict/__init__.pyi -> build/lib.linux-aarch64-3.7/multidict
    copying multidict/_multidict.c -> build/lib.linux-aarch64-3.7/multidict
    copying multidict/py.typed -> build/lib.linux-aarch64-3.7/multidict
    creating build/lib.linux-aarch64-3.7/multidict/_multilib
    copying multidict/_multilib/defs.h -> build/lib.linux-aarch64-3.7/multidict/_multilib
    copying multidict/_multilib/dict.h -> build/lib.linux-aarch64-3.7/multidict/_multilib
    copying multidict/_multilib/istr.h -> build/lib.linux-aarch64-3.7/multidict/_multilib
    copying multidict/_multilib/iter.h -> build/lib.linux-aarch64-3.7/multidict/_multilib
    copying multidict/_multilib/pair_list.h -> build/lib.linux-aarch64-3.7/multidict/_multilib
    copying multidict/_multilib/views.h -> build/lib.linux-aarch64-3.7/multidict/_multilib
    running build_ext
    building 'multidict._multidict' extension
    creating build/temp.linux-aarch64-3.7
    creating build/temp.linux-aarch64-3.7/multidict
    arm-linux-gnueabihf-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/opt/sickchill/.venv/include -I/usr/include/python3.7m -c multidict/_multidict.c -o build/temp.linux-aarch64-3.7/multidict/_multidict.o -O2 -std=c99 -Wall -Wsign-compare -Wconversion -fno-strict-aliasing -pedantic
    multidict/_multidict.c:1:10: fatal error: Python.h: No such file or directory
     #include "Python.h"
              ^~~~~~~~~~
    compilation terminated.
    error: command 'arm-linux-gnueabihf-gcc' failed with exit status 1
    ----------------------------------------
    ERROR: Failed building wheel for multidict
  Failed to build multidict
  ERROR: Could not build wheels for multidict which use PEP 517 and cannot be installed directly
  

  at .venv/lib/python3.7/site-packages/poetry/utils/env.py:1300 in _run
      1296│                 output = subprocess.check_output(
      1297│                     cmd, stderr=subprocess.STDOUT, env=env, **kwargs
      1298│                 )
      1299│         except CalledProcessError as e:
    → 1300│             raise EnvCommandError(e, input=input_)
      1301│ 
      1302│         return decode(output)
      1303│ 
      1304│     def execute(self, bin: str, *args: str, **kwargs: Any) -> Optional[int]:

The following error occurred when trying to handle this error:


  PoetryException

  Failed to install /root/.cache/pypoetry/artifacts/4d/37/78/a987ae790fd0ae8d61b254b37394e654ed19197059b0f17fd9ca2f4f82/multidict-5.1.0.tar.gz

  at .venv/lib/python3.7/site-packages/poetry/utils/pip.py:60 in pip_install
       56│                     *env.get_pip_command(),
       57│                     *args,
       58│                     env={**os.environ, "PYTHONPATH": str(env.purelib)},
       59│                 )
    →  60│         raise PoetryException(f"Failed to install {path.as_posix()}") from e
       61│ 
       62│ 
       63│ def pip_editable_install(directory: Path, environment: Env) -> Union[int, str]:
       64│     return pip_install(

  • Installing pbr (5.6.0)
  • Installing pycodestyle (2.6.0)
  • Installing pyflakes (2.2.0)
  • Installing typing-extensions (3.10.0.2)
```

At best, I'm a relatively competent novice, who is trying to piece together a solution from the snippets provided. So I think you'll understand when I say I have no idea what any of that means or how to proceed from here. 

Any suggestions for what I should do now?

**Comment by joshbaptiste** [comment]
>     multidict/_multidict.c:1:10: fatal error: Python.h: No such file or directory

This error means you need to install the development Python files for your distro .. in Debian it's called python3-dev
